### PR TITLE
tiflash: support tiflash feature branch planner_refactory

### PR DESCRIPTION
**Pull request by purelind**
* support tiflash feature branch: https://github.com/pingcap/tiflash/tree/planner_refactory

### CODE REVIEW

File: prow/config/config.yaml [hunk 1]
```diff
@@ -1369,6 +1369,16 @@ branch-protection:
...
                 strict: true
+            planner_refactory:
+              protect: true
+              required_status_checks:
+                contexts:
+                  - 'tide'
+                  - 'idc-jenkins-ci-tiflash/unit-test'
+                  - 'idc-jenkins-ci-tiflash/integration-test'
+                  - 'idc-jenkins-ci-tiflash/build'
+                  - 'license/cla'
```
Comment (SeaRise) [inline prow/config/config.yaml:1380]: Do we need to add `check-license` and `check-issue-triage-complete` here?
```suggestion
                contexts:
                  - 'tide'
                  - 'idc-jenkins-ci-tiflash/unit-test'
                  - 'idc-jenkins-ci-tiflash/integration-test'
                  - 'idc-jenkins-ci-tiflash/build'
                  - "check-issue-triage-complete"
                  - 'license/cla'
                  - "check-license"
```

Comment (purelind) [reply prow/config/config.yaml:1380]: It seems  `check-license` and `check-issue-triage-complete` only needed when brancn is `master`. other release-x.x branch also not include those two contexts

Comment (purelind) [reply prow/config/config.yaml:1380]: @Mini256 Do we need to add `check-license` and `check-issue-triage-complete` on feature branch?

Comment (SeaRise) [reply prow/config/config.yaml:1380]: Because feature branch will be merged directly to master, we should try to keep feature branch and master the same?

Comment (SeaRise) [reply prow/config/config.yaml:1380]: Because feature branch will be merged directly to master, we should try to keep feature branch and master the same?

Comment (purelind) [reply prow/config/config.yaml:1380]: got

**Review by SeaRise — APPROVED** [review]

**Comment by Mini256** [comment]
@purelind, `check-issue-triage-complete` only works on the `master` branch.

At present, `check-license` is only enabled on the `master` branch，https://github.com/pingcap/tiflash/blob/b34cd1d7c4dfe0cf3a84b21fa5c1b6e0583e0283/.github/workflows/license-checker.yml#L7

**Comment by Mini256** [comment]
BTW, maybe it is time to enable a link-issue checker for other release branches.

https://github.com/ti-community-infra/configs/blob/595523d7c6a236cdcf23c1967f510fb482859952/prow/config/external_plugins_config.yaml#L1556

**Comment by SeaRise** [comment]
> @purelind, `check-issue-triage-complete` only works on the `master` branch.
> 
> At present, `check-license` is only enabled on the `master` branch，https://github.com/pingcap/tiflash/blob/b34cd1d7c4dfe0cf3a84b21fa5c1b6e0583e0283/.github/workflows/license-checker.yml#L7

https://github.com/pingcap/tiflash/pull/4820/files#diff-f5ad8c4519812f1e04dd23ed7c7421ab44abb64df10ba188732b21e7d52a23afR6-R9

We can turn on `License checker` on the feature branch

File: prow/config/external_plugins_config.yaml [hunk 2]
```diff
@@ -1575,6 +1575,20 @@ ti-community-format-checker:
           <sub>:open_book: For more info, you can check the ["Contribute Code"](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/contribute-code.html#referring-to-an-issue) section in the development guide.</sub>
         missing_label: do-not-merge/needs-linked-issue
         skip_label: skip-issue-check
+  - repos:
+      - pingcap/tiflash
+    required_match_rules:
+      - pull_request: true
+        body: true
+        branches:
+          - planner_refactory
```
Comment (Mini256) [inline prow/config/external_plugins_config.yaml:1584]:
```suggestion
          - master
          - planner_refactory
```

And please remove line 1551.

File: prow/config/config.yaml [hunk 3]
```diff
@@ -1369,6 +1369,18 @@ branch-protection:
...
                 strict: true
+            planner_refactory:
+              protect: true
+              required_status_checks:
+                contexts:
+                  - 'tide'
+                  - 'idc-jenkins-ci-tiflash/unit-test'
+                  - 'idc-jenkins-ci-tiflash/integration-test'
+                  - 'idc-jenkins-ci-tiflash/build'
+                  - "check-issue-triage-complete"
```
Comment (Mini256) [inline prow/config/config.yaml:1380]: remove it. the issue-triage plugin only works on `master`, I think the bug issues should not be fixed on the feature branch, so it doesn't need to triage.

File: prow/config/config.yaml [hunk 4]
```diff
@@ -2039,6 +2051,21 @@ tide:
         - do-not-merge/blocked-paths
         - do-not-merge/needs-linked-issue
         - needs-rebase
+    - repos:
+        - pingcap/tiflash
+      includedBranches:
+        - planner_refactory
+      labels:
+        - status/can-merge
+        - cherry-pick-approved
```
Comment (Mini256) [inline prow/config/config.yaml:2060]: remove `cherry-pick-approved`, I think we don't need to cherry-pick PR from master to this feature branch.

**Review by Mini256 — CHANGES_REQUESTED** [review]

**Review by Mini256 — APPROVED** [review]

**Pull request merged**